src/Common/src/Interop/Unix/System.Native/Interop.GetGroupList.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        internal unsafe static uint[] GetGroupList(string userName, uint primaryGroupId)
+        {
+            Span<uint> groups = stackalloc uint[64];
```

Comment:
A pattern we sometimes use is `#if DEBUG` to set the buffer size really small (like 0 or 1) to force the growth path to be exercised. We run tests in both debug and release on CI. Up to you.